packages/styled-components/src/models/StyledComponent.js

```diff
@@ -180,6 +192,20 @@ function useStyledComponentImpl<Config: {}, Instance>(
   return createElement(elementToBeCreated, propsForElement);
 }
 
+function getShouldForwardProp(options, target, isTargetStyledComp) {
+  // $FlowFixMe
+  if (isTargetStyledComp && target.shouldForwardProp) {
+    if (options.shouldForwardProp) {
+      // compose nested shouldForwardProp calls
+      // $FlowFixMe
+      return prop => target.shouldForwardProp(prop) && options.shouldForwardProp(prop);
+    }
+    return target.shouldForwardProp;
+  }
+
+  return options.shouldForwardProp;
```

Comment:
@probablyup This can probably be done separately but we can totally add `shouldForwardProp` as a shortcut to replace the manual `withConfig`, right? Just to have alignment with Emotion.

I'd also argue now that we're touching filtering props, "transient props" (props that are always filtered) could also be revived after merging this. (Ref: #2093)